test/query.js

```diff
@@ -6,6 +6,9 @@ var Document = require(__dirname+'/../lib/document.js');
 var util = require(__dirname+'/util.js');
 var assert = require('assert');
 var Promise = require('bluebird');
+Promise.onPossiblyUnhandledRejection(function() {
```

Comment:
Can you remove this and properly catch errors?
While it may be justified, it may catch other errors that we don't want to.